VerticalMetrics/README.md

```diff
@@ -28,25 +28,39 @@ This rule can be voided if a font is being upgraded and previously had inconsist
+### 4. Win Ascent and Win Descent values must be the same as the family's tallest/deepest yMin and yMax bbox values (rounded up)
 The Microsoft [OpenType specification](https://www.microsoft.com/typography/otspec/os2.htm#wa). Recommends the following:
 >If any clipping is unacceptable, then the value should be set to yMax’.
 
 [Overpass issue #33](https://github.com/RedHatBrand/Overpass/issues/33) demonstrates what happens in MS applications when the Win Ascent and Descent are not set correctly.
 
 By changing these values, the line height will be increased in MS applications. This is can lead to very loose line heights if the bbox is exceedingly tall. This mainly occurs in families featuring Vietnamese, Devanagari and Arabic or other tall scripts. To counteract this, we enable [Use Typo Metrics](https://www.microsoft.com/typography/otspec/os2.htm#fss) and set the Typo values to match the previous Win values. By swapping the sets, we should retain the previous line heights in Windows as well as remove the clipping.
 
+We want them to be rounded up to avoid clipping at small sizes.
+
 ### 5. [Use_Typo_Metrics](https://www.microsoft.com/typography/otspec/os2.htm#fss) must be enabled
 This will force MS Applications to use the OS/2 Typo values instead of the Win values. By doing this, we can freely set the Win values to avoid clipping and control the line height with the typo values. It has the added benefit of future line height compatibility. When a new script is added, we simply change the Win values to the new yMin and yMax, without needing to worry if the line height have changed.
 
 ### 6. If the family is being updated, the line height must visually match the previous release.
 Some applications do not allow users to control the line height/leading of their fonts. Word processors and text editors are common culprits. It is essential their documents do not reflow.
 
-### 7. Hhea and Typo metrics should match
+### 7. Hhea and Typo metrics should be equal
 Hhea metrics are used in Mac OS X, whilst Microsoft uses Typo when Use_Typo_Metrics is enabled. They should ideally be identical.
 
 This rule can be voided if a font is being upgraded and previously had inconsistent values.
 
+### 8. LineGap values must be 0
+The LineGap value is a space added to the lineheight created by the union of the (typo/hhea)Ascender and (typo/hhea)Descender. It is handled differently according to the environment. It will be added under the descender height in most desktop apps, will be shared above the Ascender height and under the descender height in web browsers, and ignored in MS Word if use_typo_metrics is enabled. For better linespacing consistentcy accross platforms, (typo/hhea)linegap values must be 0.
+
+### 9. Caps should be centered if the font "main" script has uppercases (like Latin, Greek, Cyrillic)
+Web designers will thank you if you managed to have the same space above and under the uppercase letters: typoAscender-CapsHeight=-typoAscender.
```

Comment:
typo**Des**cender I presume.

sTypoAscender − sCapHeight = −sTypoDescender

or

sCapHeight − sTypoAscender = sTypoDescender

Reply:
I still question whether OS/2.sCapHeight is even used. What happens if we set sCapHeight to 0? I reckon we should be using the yMax of "Z"s bbox instead.

Reply:
sCapHeight is to the same than yMax. This is normally the position of capitals without accents or diacritics. Exactly, according to the OT specs:

> This value would normally be specified by a type designer but in situations where that is not possible, for example when a legacy font is being converted, the value may be set equal to the top of the unscaled and unhinted glyph bounding box of the glyph encoded at U+0048 (LATIN CAPITAL LETTER H). If no glyph is encoded in this position the field should be set to 0. 
> https://docs.microsoft.com/en-us/typography/opentype/spec/os2

And according to _Font Development Best Practice_:

> The “Cap height” and “x-height” values in some font design programs are not used directly by apps, but setting them correctly might be helpful, as they may be used as guidelines. Recent CSS proposals also hint that setting these correctly in the font will enable improved layout alignment for things like drop caps.
> https://silnrsi.github.io/FDBP/en-US/Line_Metrics.html

Reply:
"could be the same" is my issue. I'd prefer us to use the actual glyph bounds since this attribute is prone to being set incorrectly.

Reply:
If sCapHeight is set incorrectly, that's an issue. In the CSS3 specs:

> cap-height
> Corresponds to the top of capital letters (such as “T”, “Б”, “Σ”) in Latin, Cyrillic, Greek, etc. Calculated using sCapHeight in OpenType. 
> x-height
> Corresponds to the top of short lowercase letters (such as “m”, “л”, “α”) in Latin, Cyrillic, Greek, etc. Calculated using sxHeight in OpenType. 
> https://www.w3.org/TR/css-inline-3/

Reply:
That's a whole different issue. But yes, it should be set correctly.

Reply:
As I suspected, if I set the sCapHeight to any value, it has no influence when used to determine if the text is centred.

<img width="698" alt="Screenshot 2021-05-26 at 10 31 57" src="https://user-images.githubusercontent.com/7525512/119637917-0e14dd00-be0e-11eb-9bb8-c7d687a5cfe6.png">

_Test font: ascenders=900, descender=-200, linegaps=0, sCapHeight=400_

We need to use the "H/Z" yMax bounds.

Reply:
I don't think Rosalie meant that sCapHeight was used by applications to center text, but that the designer should use sCapHeight (or yMax of Z or H, indeed) to calculate the correct values of sTypoAscende and sTypoDescender, which are used to place the text.

And the document may say yMax of Z or H, but since this is Vertical Metrics specs, it would be good if it also says sCapHeight must be calculated correctly.

Reply:
Part of this thread has spilled over into fontbakery, https://github.com/googlefonts/fontbakery/issues/3307#issuecomment-847772912 where the formula does use this attribute.

Apologies if I'm coming across as a total jobsworth here. I really want us to make the spec less ambiguous.

Reply:
> Test font: ascenders=900, descenders=-200, linegaps=0, sCapHeight=400, yMax bbox of H=600

In your test font, it's rather yMax bbox of A=700, and this is why the text is centered: yMax of A − sTypoAscender = sTypoDescender

Reply:
> I really want us to make the spec less ambiguous.

In this regard, I still see things to change:

- "Tallest uppercase glyph (Bold Å, in this family) has a height of 1092 […] Typo Ascender = 1092. **Set to our Cap height**": the Cap height (`sCapHeight`) isn't the height of accented caps.
- "Typo Descender = -216. **Typo Ascender - upm**" : ??

Reply:
RTFM https://github.com/googlefonts/gf-docs/tree/main/VerticalMetrics#4-win-ascent-and-win-descent-values-must-be-the-same-as-the-familys-tallestdeepest-ymin-and-ymax-bbox-values

Reply:
> RTFM https://github.com/googlefonts/gf-docs/tree/main/VerticalMetrics#4-win-ascent-and-win-descent-values-must-be-the-same-as-the-familys-tallestdeepest-ymin-and-ymax-bbox-values

Indeed. So… ?